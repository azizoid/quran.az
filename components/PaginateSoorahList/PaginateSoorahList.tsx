import Link from 'next/link'

import { SOORAH_LIST } from '@/assets/soorah-list-object'
import { numberSuffixAz } from '@/utility'

export type PaginateSoorahListProps = {
  soorah: number
  translator: number
}

export const PaginateSoorahList = ({
  soorah,
  translator,
}: PaginateSoorahListProps): JSX.Element => {
  const prev = soorah === 1 ? null : soorah - 1
  const next = soorah === 114 ? null : soorah + 1

  return (
    <div className="pagination">
      {prev !== null && (
        <Link
          href={`/${prev}?t=${translator}`}
          className="pagination-item flex flex-col text-center"
        >
          {numberSuffixAz(prev)} {SOORAH_LIST[prev]['fullTitle']}
        </Link>
      )}
      <span className="pagination-disabled flex flex-col text-center">
        {numberSuffixAz(soorah)} {SOORAH_LIST[soorah]['fullTitle']}
      </span>
      {next !== null && (
        <Link
          href={`/${next}?t=${translator}`}
          className="pagination-item flex flex-col text-center"
        >
          {numberSuffixAz(next)} {SOORAH_LIST[next]['fullTitle']}
        </Link>
      )}
    </div>
  )
}
