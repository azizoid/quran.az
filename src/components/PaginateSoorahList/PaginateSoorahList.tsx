import Link from 'next/link'

import { sirasayi } from 'sirasayi'

import { SOORAH_LIST } from '@/assets/soorah-list-object'

export type PaginateSoorahListProps = {
  soorah: number
  translator: number
}

export const PaginateSoorahList = ({
  soorah,
  translator,
}: PaginateSoorahListProps): JSX.Element => {
  const prev = soorah === 1 ? null : soorah - 1
  const next = soorah === 114 ? null : soorah + 1

  return (
    <div className="pagination">
      {prev !== null && (
        <Link
          href={`/${prev}?t=${translator}`}
          className="pagination-item flex flex-col text-center"
          prefetch={false}
        >
          {sirasayi(prev)} {SOORAH_LIST[prev]['fullTitle']}
        </Link>
      )}
      <span className="pagination-disabled flex flex-col text-center">
        {sirasayi(soorah)} {SOORAH_LIST[soorah]['fullTitle']}
      </span>
      {next !== null && (
        <Link
          href={`/${next}?t=${translator}`}
          className="pagination-item flex flex-col text-center"
          prefetch={false}
        >
          {sirasayi(next)} {SOORAH_LIST[next]['fullTitle']}
        </Link>
      )}
    </div>
  )
}
