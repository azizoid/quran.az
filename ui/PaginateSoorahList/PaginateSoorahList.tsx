import Link from "next/link"
import soorah_list_object from "../../assets/soorah-list-object"

export type PaginateSoorahListProps = {
  soorah: number
  translator: number
}

export const PaginateSoorahList = ({
  soorah,
  translator,
}: PaginateSoorahListProps): JSX.Element => {
  const prev = soorah === 1 ? null : soorah - 1
  const next = soorah === 114 ? null : soorah + 1

  return (
    <div className="pagination">
      {prev !== null && (
        <Link href={`/${prev}?t=${translator}`}>
          <a className="pagination-item">
            {prev}. {soorah_list_object[prev]["fullTitle"]}
          </a>
        </Link>
      )}
      <span className="pagination-disabled">
        {soorah}. {soorah_list_object[soorah]["fullTitle"]}
      </span>
      {next !== null && (
        <Link href={`/${next}?t=${translator}`}>
          <a className="pagination-item">
            {next} {soorah_list_object[next]["fullTitle"]}
          </a>
        </Link>
      )}
    </div>
  )
}
