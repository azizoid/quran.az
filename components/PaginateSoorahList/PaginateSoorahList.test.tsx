import { render } from '@testing-library/react'
import { PaginateSoorahList } from './PaginateSoorahList'

test('PaginateSoorahList Snapshot', () => {
  const { container, getByText } = render(<PaginateSoorahList soorah={1} translator={1} />)

  expect(getByText('1-ci əl-Fatihə surəsi')).toBeInTheDocument()
  expect(getByText('2-ci əl-Bəqərə surəsi')).toBeInTheDocument()

  expect(container).toMatchSnapshot()
})

test('PaginateSoorahList Snapshot', () => {
  const { container, getByText } = render(<PaginateSoorahList soorah={114} translator={1} />)

  expect(getByText('113-cü əl-Fələq surəsi')).toBeInTheDocument()
  expect(getByText('114-cü əl-Nas surəsi')).toBeInTheDocument()

  expect(container).toMatchSnapshot()
})
