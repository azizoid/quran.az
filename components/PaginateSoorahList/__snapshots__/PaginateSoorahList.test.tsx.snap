// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginateSoorahList Snapshot 1`] = `
<div>
  <div
    class="pagination"
  >
    <span
      class="pagination-disabled flex flex-col text-center"
    >
      1-ci
       
      əl-Fatihə surəsi
    </span>
    <a
      class="pagination-item flex flex-col text-center"
      href="/2?t=1"
    >
      2-ci
       
      əl-Bəqərə surəsi
    </a>
  </div>
</div>
`;

exports[`PaginateSoorahList Snapshot 2`] = `
<div>
  <div
    class="pagination"
  >
    <a
      class="pagination-item flex flex-col text-center"
      href="/113?t=1"
    >
      113-cü
       
      əl-Fələq surəsi
    </a>
    <span
      class="pagination-disabled flex flex-col text-center"
    >
      114-cü
       
      əl-Nas surəsi
    </span>
  </div>
</div>
`;
